ServerSide/yoty/Core/Managers/Bids/BidsManager.cs

```diff
@@ -0,0 +1,291 @@
+                    await new_context.SaveChangesAsync().ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    return new Response<BuyerDTO>() { DTOObject = null, IsOperationSucceeded = false, SuccessOrFailureMessage = ex.Message };
+                }
+            }
+            //TODO dropping the dto obj from the response will save us the second db access 
+            var buyer_ent = await _context.Buyers.FindAsync(bidBuyerJoinRequest.buyerId).ConfigureAwait(false);
+            //TODO map ent to dto
+            return new Response<BuyerDTO>() { DTOObject = null, IsOperationSucceeded = true, SuccessOrFailureMessage = this.getSuccessMessage() };
+        }
+
+        public async Task<Response<SupplierProposalDTO>> AddSupplierProposal(SupplierProposalRequest supplierProposal)
+        {
+            BidEntity bid = await _context.Bids.FindAsync(supplierProposal.BidId).ConfigureAwait(false);
+            if (bid == null)
+            {
+                return new Response<SupplierProposalDTO>() { DTOObject = null, IsOperationSucceeded = false, SuccessOrFailureMessage = BidNotFoundFailString };
+            }
+            bid.CurrentOffers.Add(new SellerOfferEntity {
+                BidId = bid.Id,
+                SellerId = supplierProposal.SupplierId,
+                PublishedTime = supplierProposal.PublishedTime,
+                MinimumUnits = supplierProposal.MinimumUnits,
+                OfferedPrice = supplierProposal.ProposedPrice,
+                OfferDescription = supplierProposal.Description,
+            });
+            bid.PotenialSuplliersCounter += 1;
+            using (var new_context = new YotyContext())
+            {
+                try
+                {
+                    new_context.Bids.Update(bid);
+                    await new_context.SaveChangesAsync().ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    return new Response<SupplierProposalDTO>() { DTOObject = null, IsOperationSucceeded = false, SuccessOrFailureMessage = ex.Message };
+                }
+            }
+            //TODO dropping the dto obj from the response will save us the second db access 
+            var supplier_ent = await _context.Sellers.FindAsync(supplierProposal.SupplierId).ConfigureAwait(false);
+            //TODO map ent to dto
+            return new Response<SupplierProposalDTO>() { DTOObject = null, IsOperationSucceeded = true, SuccessOrFailureMessage = this.getSuccessMessage() };
+        }
+
+        public async Task<Response<BidDTO>> CreateNewBid(NewBidRequst productBidRequest)
+        {
+            // TODO use mapper instead
+            ProductEntity product_ent = new ProductEntity() {
+                Description = null,
+                Id = null,
+                Image = null,
+                Name = null,
+            };
+            BidEntity newBid = new BidEntity() {
```

Comment:
notice that you are creating a bid here with no product ( you initialize a product entity to all-null properties without have any assignment from the productRequest .... 

Also you can replace this conversion now with the mapper method